src/main/java/org/opensearch/knn/index/mapper/LuceneFieldMapper.java

```diff
@@ -56,31 +58,27 @@ public class LuceneFieldMapper extends KNNVectorFieldMapper {
         );
 
         this.knnMethod = input.getKnnMethodContext();
-
-        this.fieldType = new FieldType();
-
-        this.fieldType.setTokenized(false);
-        this.fieldType.setIndexOptions(IndexOptions.NONE);
-
         final SpaceType spaceType = this.knnMethod.getSpaceType();
         final VectorSimilarityFunction vectorSimilarityFunction = Optional.ofNullable(
             SPACE_TYPE_TO_VECTOR_SIMILARITY_FUNCTION.get(spaceType)
         ).orElseThrow(() -> new IllegalArgumentException(String.format("Space type [%s] is not supported for Lucene engine", spaceType)));
-
         final int dimension = input.getMappedFieldType().getDimension();
-        if (dimension > MAX_DIMENSION) {
-            throw new IllegalArgumentException(
-                String.format(
-                    "Dimension value cannot be greater than [%s] but got [%s] for vector [%s]",
-                    MAX_DIMENSION,
-                    dimension,
-                    input.getName()
-                )
-            );
```

Comment:
This is removed because validation  is taken care by Lucene itself.

Reply:
I see the rationale behind removing in-house validation. Just want to conform, do we have a contract for clients for the validation/error messages? If Lucene changes the format of the error message for validation is it ok if we return updated message from our side?

Reply:
This is a tricky thing to remove, given that lucene has the internal validations but my suggestion would be to keep it rather than removing it. Keeping the validation will have these benefits:
1. It provides us full control of the parameter value like dimensions in this case, and throw a custom error message to the user which can have more details.
2. This gives us capability of fail-fast which is always good as we will save resources here.


On other note we should also add documentation like how this value is coming and what does it controls.

Reply:
That makes sense. Yes I removed it because it looked very similar to validation done in https://github.com/apache/lucene/blob/main/lucene/core/src/java/org/apache/lucene/document/KnnVectorField.java#L38-L56, but keeping it makes sense. I will add back.

Reply:
It's actually different constant, here we do not check specifically for lucene. I think we may rename one for lucene to something like MAX_DIMENSION_LUCENE_ENGINE.

Reply:
Okay, Ill go back